src/core/models.py

```diff
@@ -112,25 +136,50 @@ def chroot_acl(self) -> str:
             - 'w' - write
             - 'a' - append
         """
-        return self.relation_data.get("chroot-acl", "cdrwa")
+        return self.relation_data.get(
+            "chroot-acl", "cdrwa"
+        )  # pyright: ignore reportGeneralTypeIssues
 
     @property
     def chroot(self) -> str:
         """The client requested root zNode path value."""
         chroot = self.relation_data.get("chroot", "")
-        if not chroot.startswith("/") and chroot:
+        if chroot and not chroot.startswith("/") and chroot:
             chroot = f"/{chroot}"
 
-        return chroot
+        return chroot  # pyright: ignore reportGeneralTypeIssues
 
 
-class ZKCluster(StateBase):
+class ZKCluster(RelationState):
     """State collection metadata for the charm application."""
 
-    def __init__(self, relation: Relation | None, component: Application, substrate: SUBSTRATES):
-        super().__init__(relation, component, substrate)
+    def __init__(
+        self,
+        relation: Relation,
```

Comment:
**todo:** Type here is `Relation | None`

Reply:
Same as https://github.com/canonical/zookeeper-operator/pull/129#discussion_r1568894833

Reply:
Duplicate, done.